### PR TITLE
feature(core): Makes several commonly-used functions public

### LINKED ISSUE #7838: Deal with private API stuff still needed by plugins
[Source](https://community.elgg.org/discussion/view/1847926/deleting-functions-marked-access-private?offset=10#elgg-object-1864622):

 - [x] Make public  elgg_view_entity_list (#7826),
 - [x] Make public get_subtype_class,
 - [x] Make public get_subtype_id,
 - [x] Make public access_get_show_hidden_status,
 - [x] Make public access_show_hidden_entities,
 - [x] Make public version of _elgg_html_decode,
 - [x] Consider what to do about elgg_get_upgrade_file_version,
 - [x] Consider what to do about generate_action_token,
 - [x] Consider what to do about _elgg_services()->hooks->getOrderedHandlers().

cc @iionly

### PR DESCRIPTION
**Pull request by mrclay**
This also adds elgg_enable_query_cache, elgg_disable_query_cache, and elgg_html_decode.

Fixes #7838

### CODE REVIEW

**Comment by ewinslow** [comment]
```
FILE: /home/travis/build/Elgg/Elgg/engine/lib/database.php
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
--------------------------------------------------------------------------------
 185 | ERROR | Missing @return tag in function comment
 196 | ERROR | Missing @return tag in function comment
--------------------------------------------------------------------------------
```

**Comment by ewinslow** [comment]
Technically a feature because it adds API?

**Comment by mrclay** [comment]
Done

File: engine/lib/output.php [hunk 1]
```diff
@@ -500,6 +500,22 @@ function _elgg_html_decode($string) {
 }
 
 /**
+ * Alias of _elgg_html_decode
```
Comment (ewinslow) [inline engine/lib/output.php]: Typo. Drop leading underscore.

**Comment by ewinslow** [comment]
LGTM. Do we need to add any tests to cover any of these, or are the tests already there?

**Comment by mrclay** [comment]
Probably missing tests.

File: engine/lib/views.php [hunk 2]
```diff
@@ -954,7 +954,6 @@ function elgg_view_annotation(\ElggAnnotation $annotation, array $vars = array()
  *      'no_results'       Message to display if no results (string|Closure)
  *
  * @return string The rendered list of entities
- * @access private
  */
 function elgg_view_entity_list($entities, array $vars = array()) {
```
Comment (mrclay) [inline engine/lib/views.php]: @iionly it's now public

Comment (iionly) [inline engine/lib/views.php]: Sorry. I had missed it.

**Pull request merged**